src/lib/files/YarnLinkedFilesConfig.ts

```diff
@@ -0,0 +1,134 @@
+import { lstat, readdir, realpath, Stats } from 'fs-extra'
+import * as glob from 'globby'
+import { dirname, join } from 'path'
+import { partition, unnest } from 'ramda'
+import log from '../../logger'
+
+const isScopedDirOrLink = (path: string, stats: Stats) => {
+  return stats != null && ((path.startsWith('@') && stats.isDirectory()) || stats.isSymbolicLink())
+}
+
+const isLink = (_: string, stats: Stats) => {
+  return stats != null && stats.isSymbolicLink()
+}
+
+export class YarnSymlinkedModulesConfig {
+  public static async createConfig(projectSrc: string) {
+    const conf = new YarnSymlinkedModulesConfig(projectSrc)
+    await conf.init()
+    return conf
+  }
+
+  private stack = []
+  private graph: Record<string, string[]> = {}
+  private _metadata: Record<string, string> = {}
+
+  constructor(private projectSrc: string) {}
+
+  get metadata() {
+    return this._metadata
+  }
+
+  get symlinkedDepsDirs() {
+    return Object.values(this.metadata)
+  }
+
+  get symlinkedModules() {
+    return Object.keys(this.metadata)
+  }
+
+  get symlinkedDependencies() {
+    return Object.keys(this.metadata).map((moduleName: string) => {
+      return {
+        moduleName,
+        path: this.metadata[moduleName],
+      }
+    })
+  }
+
+  public async init() {
+    const allPackageJsonsFolders = (await glob([join('*', 'package.json')], { cwd: this.projectSrc })).map(dirname)
+    this.stack.push(...allPackageJsonsFolders)
+    while (this.stack.length > 0) {
+      const moduleFolder = this.stack.pop()
+      const dependencies = await this.discoverDependencies(moduleFolder, this.projectSrc)
+      this.graph[moduleFolder] = dependencies
+      this.addSubDependenciesToStack(dependencies)
+    }
+  }
+
+  public toJson() {
+    return JSON.stringify({
+      metadata: this.metadata,
+      graph: this.graph,
+    })
+  }
+
+  private async discoverDependencies(currentModule: string, projectSrc: string): Promise<string[]> {
```

Comment:
`projectSrc` is an instance variable. You don't need to add this as a parameter of the function, access it with `this.projectSrc`